control-plane/pkg/reconciler/consumergroup/controller.go

```diff
@@ -116,25 +114,26 @@ func enqueueConsumerGroupFromConsumer(enqueue func(name types.NamespacedName)) f
 	}
 }
 
-func createKafkaSourceScheduler(ctx context.Context, c SchedulerConfig, kafkaLister sources.KafkaSourceLister) scheduler.Scheduler {
+func createKafkaScheduler(ctx context.Context, c SchedulerConfig, ssName string) scheduler.Scheduler {
+	lister := consumergroup.Get(ctx).Lister()
 	return createStatefulSetScheduler(
 		ctx,
 		SchedulerConfig{
-			StatefulSetName:     "kafka-source-dispatcher",
+			StatefulSetName:     ssName,
 			RefreshPeriod:       c.RefreshPeriod,
 			Capacity:            c.Capacity,
 			SchedulerPolicyType: c.SchedulerPolicyType,
 			SchedulerPolicy:     c.SchedulerPolicy,
 			DeSchedulerPolicy:   c.DeSchedulerPolicy,
 		},
 		func() ([]scheduler.VPod, error) {
-			kafkaSources, err := kafkaLister.List(labels.Everything())
+			consumerGroups, err := lister.List(labels.Everything())
```

Comment:
This lister needs to be on a per top-level resource basis because otherwise, we consider every consumer group for each statefulset, we can filter based on the owner ref or we can add a label with the kind associated with a ConsumerGroup.

/hold

Reply:
Thanks!! @pierDipi I added labels to CGs